### PR TITLE
js2 homework week2

### PR DESCRIPTION
**Pull request by STEP0HEN**
added js2 wk 2 homework

### CODE REVIEW

File: javascript/javascript2/week2/js2-wk2-homework/doubling Numbers/doublingOfNumber.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+"use strict"
+
+let numbers = [1, 2, 3, 4];
+const newNumbers = numbers.filter(OddNumbers => OddNumbers % 2 !== 0).map(multiplyOddNumbers => multiplyOddNumbers * 2);
```
Comment (aleksd-dfds) [inline javascript/javascript2/week2/js2-wk2-homework/doubling Numbers/doublingOfNumber.js]: variable naming - the number is not odd HERE, so this naming is super confusing. Besides it is ONE number, not many. So simply using `x`, `y` or `num` is actually better here. Same for `multiplyOddNumbers`, it is not an ACTION, it is a NUMBER. Sorry for the caps lock, that was for emphasis.

File: javascript/javascript2/week2/js2-wk2-homework/workingWithMovies/workingWithMovies.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
...grand bleu","year": 1988,"rating": 7.6,"votes": 41252,"running_times": 10080},{"title": "Le locataire","year": 1976,"rating": 7.8,"votes": 30919,"running_times": 7560},{"title": "Le mépris","year": 1963,"rating": 7.7,"votes": 22380,"running_times": 6180},{"title": "Le procès","year": 1962,"rating": 7.8,"votes": 15401,"running_times": 7140},{"title": "Le salaire de la peur","year": 1953,"rating": 8.2,"votes": 40345,"running_times": 9360},{"title": "Le violon rouge","year": 1998,"rating": 7.7,"votes": 28394,"running_times": 8400},{"title": "Lean on Me","year": 1989,"rating": 7.4,"votes": 15573,"running_times": 6480},{"title": "Leap Year","year": 2010,"rating": 6.5,"votes": 78668,"running_times": 6000},{"title": "Leap of Faith","year": 1992,"rating": 6.1,"votes": 10526,"running_times": 6480},{"title": "Leatherface: Texas Chainsaw Massacre III","year": 1990,"rating": 5.2,"votes": 11025,"running_times": 5100},{"title": "Leatherheads","year": 2008,"rating": 6,"votes": 28907,"running_times": 6840},{"title": "Leaves of Grass","year": 2009,"rating": 6.5,"votes": 24966,"running_times": 6300},{"title": "Leaving Las Vegas","year": 1995,"rating": 7.6,"votes": 100091,"running_times": 6660},{"tit
```
Comment (aleksd-dfds) [inline javascript/javascript2/week2/js2-wk2-homework/workingWithMovies/workingWithMovies.js]: `movies.filter(movies => movies.title.length <= 3)` You're using the same naming for the array as for the elements of the array, that is very confusing. The variable inside the filter should be called `movie` or `aMovie`

Comment (aleksd-dfds) [inline javascript/javascript2/week2/js2-wk2-homework/workingWithMovies/workingWithMovies.js]: same here

Comment (aleksd-dfds) [inline javascript/javascript2/week2/js2-wk2-homework/workingWithMovies/workingWithMovies.js]: same here

Comment (aleksd-dfds) [inline javascript/javascript2/week2/js2-wk2-homework/workingWithMovies/workingWithMovies.js]: should you be returning `ratingBasedTag.tag`? or just `ratingBasedTag`? I would do the second one. So you do `ratingBasedTag.tag='Good'` here but don't return. And then at the end of the function return the whole object.

Comment (aleksd-dfds) [inline javascript/javascript2/week2/js2-wk2-homework/workingWithMovies/workingWithMovies.js]: filter variable naming unclear, again.

Comment (aleksd-dfds) [inline javascript/javascript2/week2/js2-wk2-homework/workingWithMovies/workingWithMovies.js:47]: you can shorten the whole if statement to: `return (moviesSplit.includes('surfer') || moviesSplit.includes('alien') || moviesSplit.includes('benjamin'));` Doing `return true` and `return false` is unncessary.

Comment (aleksd-dfds) [inline javascript/javascript2/week2/js2-wk2-homework/workingWithMovies/workingWithMovies.js]: what's going on here? What are you getting the .length of? You already returned. Redo this part, as it doesn't make sense.

Comment (aleksd-dfds) [inline javascript/javascript2/week2/js2-wk2-homework/workingWithMovies/workingWithMovies.js]: This is a confusing solution, but it maybe it works. Did you test it? What does it mean to do a double arrow function a => b => c? 
That's what you have here.

**Review by aleksd-dfds — COMMENTED** [review]
Overall good effort, sometimes variable naming inside the filter or map functions should be clearer. The duplicate exercise should be redone to be clearer and simpler.

**Comment by STEP0HEN** [comment]
Thanks for the review Aleks, I've implemented all the changes.

**Pull request merged**